DiscordBot/report.py

```diff
@@ -118,124 +106,107 @@ async def handle_message(self, message):
-                                                 self.NUDITY_OPTIONS)
-            if i == 1:
-                self.state = State.REPORT_COMPLETE
-                reply = self.REPORT_COMPLETE_OTHER_MESSAGE
+            self.state = State.AWAITING_FINAL_ADDITIONAL_INFORMATION
+            reply = f"Please add any additional information you think is relevant ({self.EXPLANATION_INPUT_LIMIT} word limit)."
             return [reply]
         
         if self.state == State.AWAITING_EXPLANATION_INPUT:
             if len(message.content.split()) > self.EXPLANATION_INPUT_LIMIT:
                 reply = f"Please do not exceed the {self.EXPLANATION_INPUT_LIMIT} word limit."
             else:
-                # TODO: forward explantation to moderator?
-                reply = self.REPORT_COMPLETE_OTHER_MESSAGE
-                self.state = State.REPORT_COMPLETE
+                # TODO: attach explanation to moderator report
+                self.state = State.AWAITING_FINAL_ADDITIONAL_INFORMATION
+                reply = f"Please add any additional information you think is relevant ({self.EXPLANATION_INPUT_LIMIT} word limit)."
             return [reply]
         
-        if self.state == State.AWAITING_MINOR_EXPLANATION_INPUT:
+        if self.state == State.AWAITING_NUDITY_EXPLANATION_INPUT:
             if len(message.content.split()) > self.EXPLANATION_INPUT_LIMIT:
                 reply = f"Please do not exceed the {self.EXPLANATION_INPUT_LIMIT} word limit."
             else:
-                # TODO: forward explantation to moderator?
-                self.state = State.AWAITING_MET_IN_PERSON_ANSWER
-                reply = self.create_options_list("Have you or the person you are reporting on behalf met them in person?",
+                # TODO: attach explanation to moderator report
+                self.state = State.AWAITING_MINOR_INVOLVEMENT_ANSWER
+                reply = self.create_options_list("Does the abuse involve someone under 18, either you or someone else?",
                                             self.YES_NO_OPTIONS)
             return [reply]
+        
+        if self.state == State.AWAITING_FINAL_ADDITIONAL_INFORMATION:
+            if len(message.content.split()) > self.EXPLANATION_INPUT_LIMIT:
+                reply = f"Please do not exceed the {self.EXPLANATION_INPUT_LIMIT} word limit."
+            else:
+                self.state = State.AWAITING_BLOCK_ANSWER
+                reply = [self.REPORT_COMPLETE_SEXTORTION_MESSAGE,
+                        self.create_options_list("Would you like to block this account?",
+                                                 self.YES_NO_OPTIONS)]
+            return reply
+        
+        if self.state == State.AWAITING_BLOCK_ANSWER:
+            i = self.get_index(message, self.YES_NO_OPTIONS)
+            if i == -1:
+                return ["Please enter a number corresponding to the given options."]
+            if i == 0:
+                # TODO: yes, block account
```

Comment:
Could we add a message here saying something like, "The account you've reported will be blocked" ? Might better signpost the blocking that will happen on the moderator, especially if we haven't fleshed out the block action yet